src/business_logic/transaction/objects/internal_deploy.rs

```diff
@@ -0,0 +1,202 @@
+    ) -> Result<TransactionExecutionInfo, StarkwareError> {
+        if self.constructor_calldata.is_empty() {
+            return Err(StarkwareError::TransactionFailed);
+        }
+
+        let class_hash: [u8; 32] = self.contract_hash;
+        let call_info = CallInfo::empty_constructor_call(
+            self.contract_address.clone(),
+            Address(0.into()),
+            Some(class_hash),
+        );
+
+        let resources_manager = ExecutionResourcesManager {
+            ..Default::default()
+        };
+
+        let changes = state.count_actual_storage_changes();
+        let actual_resources = calculate_tx_resources(
+            resources_manager,
+            &[Some(call_info.clone())],
+            self.tx_type.clone(),
+            changes,
+            None,
+        )
+        .map_err(|_| StarkwareError::UnexpectedHolesL2toL1Messages)?;
+
+        Ok(
+            TransactionExecutionInfo::create_concurrent_stage_execution_info(
+                None,
+                Some(call_info),
+                actual_resources,
+                Some(self.tx_type.clone()),
+            ),
+        )
+    }
+
+    pub fn invoke_constructor<S: Default + State + StateReader + Clone>(
+        &self,
+        state: &mut S,
+        general_config: StarknetGeneralConfig,
+    ) -> Result<TransactionExecutionInfo, ExecutionError> {
+        let entry_point_selector = felt_str!(
+            "1159040026212278395030414237414753050475174923702621880048416706425641521556"
+        );
+        let call = ExecutionEntryPoint::new(
+            self.contract_address.clone(),
+            self.constructor_calldata.clone(),
+            entry_point_selector,
+            Address(0.into()),
+            EntryPointType::Constructor,
+            None,
+            None,
+        );
+
+        let tx_execution_context = TransactionExecutionContext::new(
+            Address(Felt::zero()),
+            self.hash_value.clone(),
+            Vec::new(),
+            0,
+            Felt::zero(),
+            general_config.invoke_tx_max_n_steps,
+            self.version,
+        );
+
+        let mut resources_manager = ExecutionResourcesManager::default();
+        let call_info = call.execute(
+            state,
+            &general_config,
+            &mut resources_manager,
+            &tx_execution_context,
+        )?;
+
+        let changes = state.count_actual_storage_changes();
+        let actual_resources = calculate_tx_resources(
+            resources_manager,
+            &[Some(call_info.clone())],
+            self.tx_type.clone(),
+            changes,
+            None,
+        )?;
+
+        Ok(
+            TransactionExecutionInfo::create_concurrent_stage_execution_info(
+                None,
+                Some(call_info),
+                actual_resources,
+                Some(self.tx_type.clone()),
+            ),
+        )
+    }
+
+    pub fn execute<S: Default + State + StateReader + Clone>(
```

Comment:
Add some docs to this method specifying what it does. The same with the apply method